### PR TITLE
Show Client::join error on join failure

### PR DESCRIPTION
**Pull request by toyota-corolla0**

### CODE REVIEW

File: src/components/app.rs [hunk 1]
```diff
@@ -109,7 +109,12 @@ pub fn App(cx: Scope) -> impl IntoView {
               fallback=|_| empty_view()
             >
               {move || view!{ cx, <p>{
-                format!("Failed to join federation: {:?}", join_action_error.with(|e| anyhow!("{:?}", e)))
+                format!("Failed to join federation: {:?}", join_action_error.with(|e_opt| {
+                  match e_opt {
+                    Some(e) => anyhow!("{:?}", e),
+                    None => anyhow!("")
+                  }
+                }))
```
Comment (elsirion) [inline src/components/app.rs]: Not sure this is the most elegant way to achieve what you want, but works for webimint

File: src/components/app.rs [hunk 2]
```diff
@@ -32,34 +32,37 @@ pub fn App(cx: Scope) -> impl IntoView {
...
     let show_wallet = move || {
         let select_wallet = select_wallet_action.value().get();
         select_wallet.is_some()
-            && (join_action.value().get() == Some(Some(())) || select_wallet == Some(Some(true)))
+            && (join_action.value().with(|r| {
+                if let Some(Ok(_)) = r {
+                    true
+                } else {
+                    false
+                }
```
Comment (elsirion) [inline src/components/app.rs:63]: This could use [`matches!`](https://doc.rust-lang.org/std/macro.matches.html)

**Review by elsirion — APPROVED** [review]
Thx for the PR!

In other places we handle errors a bit more leniently: show the message but don't hide the input field so users can retry.

Screenshot 2023-11-15 at 13-14-08 Fedimint Web Client

The error messages are typically wrong (most of the time the client didn't crash but returned an error we didn't handle afaik), but the way they are being displayed yould be used as inspiration.

**Comment by toyota-corolla0** [comment]
@elsirion Thanks for the help! I've addressed these issues in #44.

**Pull request merged**